File: SmartPocketAPI/Models/Movement.cs
using SmartPocketAPI.Models.General;

namespace SmartPocketAPI.Models;

public class Movement : ITimestampedEntity
{
    public int Id { get; set; }
    public DateOnly MovementDate { get; set; }
    public string Description { get; set; }
    public decimal Amount { get; set; }
    public int? InstallmentNumber { get; set; }

    public Guid UserId { get; set; }
    public User User { get; set; }

    public int CategoryId { get; set; }
    public Category Category { get; set; }

    public int PaymentMethodId { get; set; }
    public PaymentMethod PaymentMethod { get; set; }

    public int? RecurringPaymentId { get; set; }
    public RecurringPayment RecurringPayment { get; set; }

    public int MovementTypeId { get; set; }
    public MovementType MovementType { get; set; }

    public int? CreditCardPaymentId { get; set; }
    public PaymentMethod CreditCardPayment { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

internal record MovementRecord(string Description, bool Pending, string Category, decimal Amount, DateOnly MovementDate);
public record MovementDocument(string Card, DateOnly Date, string Concept, decimal Amount);

File: SmartPocketWeb2/src/app/models/movements/movement.ts
import { User } from "../auth/user";
import { Category } from "./category";
import { MovementType } from "./movement-type";
import { PaymentMethod } from "./payment-method";
import { RecurringPayment } from "./recurring-payment";

export interface Movement {
    id: number;
    movementDate: Date;
    description: string;
    amount: number;
    installmentNumber?: number;
    userId: string;
    user: User;
    categoryId: number;
    category: Category;
    paymentMethodId: number;
    paymentMethod: PaymentMethod;
    recurringPaymentId?: number;
    recurringPayment: RecurringPayment;
    movementTypeId: number;
    movementType: MovementType;
    creditCardPaymentId?: number;
    creditCardPayment: PaymentMethod;
    createdAt: Date;
    updatedAt: Date;
}